python/cudf/cudf/core/udf/strings_typing.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import operator
+
+from numba import types
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_decl_registry
+
+from strings_udf._typing import (
+    StringView,
+    bool_binary_funcs,
+    id_unary_funcs,
+    int_binary_funcs,
+    size_type,
+    string_view,
+)
+
+from cudf.core.udf import masked_typing
+from cudf.core.udf._ops import comparison_ops
+from cudf.core.udf.masked_typing import MaskedType
+
+masked_typing.MASKED_INIT_MAP[types.pyobject] = string_view
+masked_typing.MASKED_INIT_MAP[string_view] = string_view
+
+
+def _is_valid_string_arg(ty):
+    return (
+        isinstance(ty, MaskedType) and isinstance(ty.value_type, StringView)
+    ) or isinstance(ty, types.StringLiteral)
+
+
+def register_string_function(func):
```

Comment:
This file looks great now. Significantly more readable IMO. I would recommend adding docs to all the helper functions/decorators. Nothing too extensive, just a brief description is fine since these are all internal.